src/anomaly/metaml.rs

```diff
@@ -0,0 +1,214 @@
+/// * The data at row `i` of the features data corresponds to the data at row `i` of the targets data
+pub struct MetaMLDataset {
+    pub features: Array2<f32>,
+    pub targets: Array1<f32>,
+}
+
+impl MetaMLDataset {
+    /// Creates a dataset for training a meta-ml model from a set of feature values
+    /// and their corresponding target values
+    ///
+    /// # Panics
+    /// * If the number of columns in the features data isn't 6
+    /// * If the number of rows in the features data doesn't match the number of
+    /// elements in the targets data
+    ///
+    /// # Examples
+    /// ```no_run
+    /// # use ndarray::{Array1, Array2};
+    /// # use clam::anomaly::metaml::MetaMLDataset;
+    /// #
+    /// let features: Array2<f32> = unimplemented!();
+    /// let targets: Array1<f32> = unimplemented!();
+    ///
+    /// // Remember features must have 6 columns, and the same number of rows as targets
+    /// assert!(features.ncols() == 6);
+    /// assert!(features.nrows() == targets.len());
+    ///
+    /// let dataset: MetaMLDataset = MetaMLDataset::new(features, targets);
+    ///
+    /// // Do something with the dataset
+    /// ```
+    pub fn new(features: Array2<f32>, targets: Array1<f32>) -> Self {
+        // Double-check meta-information about the data
+        assert!(features.ncols() == 6);
+        assert!(features.nrows() == targets.len());
+
+        MetaMLDataset { features, targets }
+    }
+}
+
+/// Creates a dataset for training a meta-ml model from input data on disk.
+///
+/// # Panics
+/// * If either of the given files can't be found, opened, or parsed as `f32`s
+/// * If the data contained within the features file isn't two-dimensional
+/// * If the data contained within the targets file isn't one-dimensional
+/// * If the number of columns in the features data isn't 6
+/// * If the number of rows in the features data doesn't match the number of
+/// elements in the targets data
+///
+/// # Examples
+/// ```no_run
+/// # use clam::anomaly::metaml::create_dataset_from_npy;
+/// # use clam::anomaly::metaml::MetaMLDataset;
+/// #
+/// // File paths to training data in the numpy .npy file format
+/// let features_path = "path/to/data/features.npy";
+/// let targets_path = "path/to/data/targets.npy";
+///
+/// // Construct a dataset from the features and targets files
+/// let dataset: MetaMLDataset = create_dataset_from_npy(features_path, targets_path);
+///
+/// // Do something with the dataset
+/// todo!();
+/// ```
+pub fn create_dataset_from_npy(features_file_path: &str, targets_file_path: &str) -> MetaMLDataset {
+    // Read in the training data
+    let features_f64: Array2<f64> = read_npy(features_file_path).expect("failed to read the features data file");
+    let targets_f64: Array1<f64> = read_npy(targets_file_path).expect("failed to read the outputs data file");
+
+    // Transform the training data from f64 to f32 (we are given f64, but automl uses f32s)

```

Comment:
This is the second part of the two-part advice... this comment is **very** useful.